near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,298 @@
+
+/// This file makes heavy use of Copy-on-write references to strings.
+/// Since the logs should be light weight they should not clone any string.
+/// Using generic arguments bound to `Into<Cow<'a, str>>` any reference passed that uses
+/// can provide a reference to a string can be used. E.g. `&String, String, &str, &AccountId`
+/// This also makes it more ergonomic since not only one type can be passed.
```

Comment:
Okay so I'm not seeing why you need `Cow` for this. The negative for this is primarily the added code size for introducing a new part of std but also some unnecessary compute cycles. Also, for these logs, there is never any need to deserialize them within the contract.

You can remove the deserialize implementation altogether because these logs will only be parsed off-chain, and just use references for these things, no?

Reply:
@austinabell  I just redid it to use `&'a str` directly since the lifetimes of the logs are shorter than the caller's.